tagstudio/src/qt/modals/fix_unlinked.py

```diff
@@ -125,23 +123,19 @@ def refresh_missing_files(self):
             maximum=self.lib.entries_count,
         )
 
-        pw.show()
-
-        iterator = FunctionIterator(self.tracker.refresh_missing_files)
-        iterator.value.connect(lambda v: pw.update_progress(v + 1))
-        r = CustomRunnable(iterator.run)
-        QThreadPool.globalInstance().start(r)
-        r.done.connect(
-            lambda: (
-                pw.hide(),
-                pw.deleteLater(),
-                self.set_missing_count(self.tracker.missing_files_count),
-                self.delete_modal.refresh_list(),
-            )
+        pw.from_iterable_function(
+            self.tracker.refresh_missing_files,
+            None,
+            self.set_missing_count,
+            self.delete_modal.refresh_list,
         )
 
-    def set_missing_count(self, count: int):
-        self.missing_count = count
+    def set_missing_count(self, count: int | None = None):
+        if count:
```

Comment:
shouldnt the condition be like this? Otherwise it will be misbehaving when the value is `0`
```suggestion
        if count is not None:
```